### ISSUE TITLE
web_responsive scroll bug

### ISSUE THREAD

**Issue by zefoo** [OCA/web#1182]
Grateful for the responsive plugin! It almost seems to work faster than the Odoo native version (somehow)... I'm liking it. 

I have a custom model with a tree view in the app icon slide out and it seems to work ok but I cannot actually scroll down on the tree view. Do I need to add something somewhere to get it to scroll down on mobile? Works fine on desktop. It does work well with contacts and tickets and what not so I'm thinking there is something specific with my own tree view that is causing this. 

Version 11. 

Thank you!

**Comment by pedrobaeza** [comment]
Without some screenshots/video and maybe the code of the view is difficult to see what is the problem.

**Comment by zefoo** [comment]
Thank you Pedro for your quick response! 

Here's a screencast from my iPhone on what it looks like: https://www.dropbox.com/s/dmhgmw9a8nxexg1/Video%20Feb%2007%2C%205%2008%2015%20PM.mp4?dl=0

I've had audio codec issues with the iPhone screencast functionality... not sure if audio will come through. In short, you can get a sense for what's happening and how it's not scrolling down. Code-wise, nothing special here - standard model with a tree view. I can't imagine anything out-of-the-ordinary there but if you think that would be useful I can assemble that. About to step out for a few. 

Also, in the event you're on freenode I can shoot you a message there if it'd help. Appreciate any time you spend looking into this!

**Comment by pedrobaeza** [comment]
OK, I see the problem. That's not happening in other browsers AFAIK. Can you try in Chrome or Firefox? Any way, the ideal thing is to provide a kanban view for the model for being automatically used when on mobile view (like sales orders for example).

**Comment by zefoo** [comment]
Thank you for your quick response, err, not sure why I wasn't notified on this. 

To be more clear, this works fine in Safari desktop. I originally recorded screencast on Safari mobile. I do, however, still see the issue on Chrome mobile. 

We have many companies in the system and from a space perspective, Kanban view doesn't seem too feasible/practical. (eg. it'd take up too much unnecessary space on the small screen, I believe)

Also, for the record, one of these tree views (which happens with all tree views not part of a stock tree view such as contacts), has issues per my screencast. Issues being someone cannot scroll down. Some of these tree views simply extend res.partner, too. Also, I recorded the screencast from an iPhone 6S on iOS 12.1.3. 

To recap: The main issue seems to be if I add a tree view of my own, a user cannot seem to scroll down past the bottom of the page in mobile Safari and Chrome. I'm wondering if I need to apply the mobile responsive to a tree view somehow?

Everything desktop-wise works great, still testing mobile. Thank you!

**Comment by pedrobaeza** [comment]
About kanban view, I had to say that the advantages are more than disadvantages, as in tree view you have to continuosly scroll left/right. Although it's true they need more vertical space, it's more natural to scroll up/down to see data.

About the custom tree views, there's nothing special to apply for responsive, but it seems your custom views have something different from standard ones that makes this to fail. Without seeing the view arch, I can't say more.

**Comment by zefoo** [comment]
Aha, I see what you're saying. I can look into Kanban as a last resort but that's likely not an option. Appreciate you elaborating on your point, thank you.

Here's the code:

```
    <record model="ir.ui.view" id="company_list_view">
      <field name="name">res.partner.companies.tree</field>
      <field name="model">res.partner</field>
      <field name="arch" type="xml">
        <tree create="false">
          <field name="name"/>
          <field name="phone"/>
          <field name="city"/>
          <field name="state_id"/>
          <field name="user_id"/>
        </tree>
      </field>
    </record>

    <record model="ir.actions.act_window" id="company_list_action">
      <field name="name">Companies</field>
      <field name="type">ir.actions.act_window</field>
      <field name="res_model">res.partner</field>
      <field name="view_id" ref="company_list_view"/>
      <field name="search_view_id" ref="company_list_search_view"/>
      <field name="context" eval="{'form_view_ref': 'companies.company_form_view', 'default_country_id': ref('base.us'), 'default_is_company': '1'}"/>
      <field name="view_mode">tree,form</field>
      <field name="domain">[('is_company', '=', 1), ('commercial_partner_id', '!=', 1)]</field>
    </record>

    <menuitem name="Companies" id="companies.menu_root" sequence="10"/>
    <menuitem name="Companies 2" id="companies.menu_companies_list" parent="companies.menu_root"
              action="company_list_action" sequence="5"/>
```

If there's anything else I can share let me know otherwise just another tree view from my vantage point, but perhaps you see something more? I'm open to any more testing too or if there's something you know I could check. 

Grateful for your time!

**Comment by pedrobaeza** [comment]
Please try removing the lines `<field name="view_id"...` and `<field name="context"...`.

**Comment by zefoo** [comment]
Now it looks like this:

```
    <record model="ir.actions.act_window" id="company_list_action">
      <field name="name">Companies</field>
      <field name="type">ir.actions.act_window</field>
      <field name="res_model">res.partner</field>
      <!--<field name="view_id" ref="company_list_view"/>-->
      <field name="search_view_id" ref="company_list_search_view"/>
      <!--<field name="context" eval="{'form_view_ref': 'companies.company_form_view', 'default_country_id': ref('base.us'), 'default_is_company': '1'}"/>-->
      <field name="view_mode">tree,form</field>
      <field name="domain">[('is_company', '=', 1), ('commercial_partner_id', '!=', 1)]</field>
    </record>
```

Restarted Odoo. Same issue on iOS in mobile Chrome and Safari. (I was using mobile Safari incognito too and ensured caches were cleared). 

I've done a bit of mobile dev, too. This seems strange, as if the page is stuck and doesn't allow scrolling down. I can't reproduce this issue on the desktop and spot any errors. I wonder if there is a meta tag for mobile viewport that would cause this but can't imagine that... have you ever seen this before? If anything, I know it happens on only my own custom views so it'd seem like some CSS isn't getting applied. And even then, removing view_id may use default views (but I'm relatively newer to Odoo and still learning the nuances). I haven't modified much of the actual HTML templates or anything like that which is also strange on this view. 

Any other things I can test/test or I can share? Thank Pedro! Grateful for your thoughts (and baffled why I'm not getting e-mail notifications of this thread. I tried to unsubscribe and subscribe again to see if that works)

**Comment by zefoo** [comment]
I suspect this has to be CSS related, I can't imagine it's javascript related. One test out of the box could be to see if this issue exists on a fresh Odoo 11 install and re-using res.partner on a new tree view under a new nav item or such.

**Comment by pedrobaeza** [comment]
@zefoo several things:

* It's not enough to restart Odoo. You have to update the module.
* For this specific case, as you are commenting some fields of the model, they are not going to be reset, so you have to explicitly put them as empty or directly remove the act_window record in your DB and the update again the module.

**Comment by zefoo** [comment]
To be more clear, I am force updating the module. I ain't that noob. ;)

I can try renaming things, per your suggestion... although I think the next test (if that doesn't work) would be to get you a view on a fresh Odoo 11 install with web_responsive so we can reproduce this. Agree?

**Comment by pedrobaeza** [comment]
Deal

**Comment by zefoo** [comment]
Alrighty, finally put this to the test. Sure enough, it does seem reproducible and works incorrectly like I thought on test data. Hopefully this helps. I've done what I can to make this as easy as possible. 

Here's a ZIP file: https://www.dropbox.com/s/ku33hmm6tl9xre5/odoo-11-mobile-test.zip?dl=0 ~325MB

In it, is the Odoo 11 (latest stable) files. In custom/ I stuck the latest stable web_responsive (for 11) as well as a mobile_test module folder as well as a quick ./start.sh (which is not primarily how I start odoo but did it for kicks).

I believe the steps to reproduce are:

1. Download zip
2. Extract zip
3. Oh, you'll need to create a quick venv – I'm using python 3.6
4. Activate venv
5. Configure local postgres connection
6. Start Odoo (with ./start or something else)
7. Load database (it's in custom/): mobile_test1_2019-02-27_19-07-39.zip
8. Username is mobile_test1@localhost.com pass is admin
9. If you load up odoo and select database it should have mobile_test module already added... in which case, you should see this issue if you go to "Mobile Test" from the nav from mobile phone (eg. with test data, you can't scroll down, per above video)

Instead of ngrok I like: https://serveo.net – to get to this from iPhone or other mobile phone (I haven't tested in Android)

In the event you don't want to download Odoo 11 + database and simply want to download a very simple (small) plugin, and you have Odoo 11 handy + latest web_responsive + test data (or data in res.partner), here's the simple module only which I believe will reproduce same issue: 

https://www.dropbox.com/s/qc9yzik526wwghp/mobile_test.zip?dl=0

I'm really curious what you find and hope you see the same thing as I. I suspect this is relatively simple fix, likely adding a class or getting web_responsive to account for other nav items/pages/act_windows (maybe a css class or ID we need to add somewhere?). 

I am grateful for your time in looking into this, thank you! 

PS. I realize my verbosity in how to set this up is likely not needed. :)

**Comment by pedrobaeza** [comment]
OK, I see that you are not doing anything special regarding the view, but relying on standard `res.partner` views, so I suppose this is reproducible in a runbot session.

@Yajo can you figure out what can be the problem with this?

**Comment by zefoo** [comment]
@pedrobaeza Correct. While I do have my own views in my own models where this is an issue too, I suspect there's something else that's needed for views in general... specifically, maybe the current responsive CSS that allows for scrolling is too strict and we need to loosen it up so it applies to all tree views or such.

Thank you for taking a look!

**Comment by yajo** [comment]
I have absolutely no idea on what can that be, but I guess that somebody with an iphone should debug it and propose a PR to fix it.

If we're able to reproduce in runbot, it's always easier to debug.

**Comment by zefoo** [comment]
One thing I don't think I've communicated. web_responsive is broke out-of-the-box on tree views. eg. I know my original mentions was how it's broke on a tree view from a custom module, but so we'e clear (which I haven't shared), web_responsive is broke on *all* tree views. Even contacts demo data, if I flip to tree view from kanban, it doesn't allow scrolling past whatever is in the viewport. I wonder if tree views are using some sort of special CSS-fu that's preventing this? Maybe positioning or such. I can't imagine this is all javascript related. Hmm.

**Comment by yajo** [comment]
I tested on firefox & chrome, desktop & mobile. All work fine. Possibly it's some limitation in Safari. Let's hope some i-thing owner can fix this.

**Comment by zefoo** [comment]
Yajo, thank you. Yes, seems fine on desktop Safari and several browsers. Out of curiosity, what mobile phone did you test on? Android? Chrome browser I assume? 

Good to know it's iPhone specific on Safari.

**Comment by yajo** [comment]
I tested on Android, with Chrome and Firefox for Android. AFAIK, iOS chrome uses the same webkit engine as Safari, so you should experience the same bugs. As a workaround, you can try Firefox for iOS, which I think uses a different engine.

**Comment by zefoo** [comment]
The solution here was to set up Kanban and use that instead of tree view. Desktop view is tree view, mobile view is kanban. I believe this was original intention. 

With custom kanban res.partner, I was seeing strange issues like "field doesn't exist" (aka. street, street2, city, etc)... very strange. I was seeing this with safari browser window as narrow as possible (eg. enabling responsive stuff), *only* with default user agent set to iPhone, with *and* without web-responsive module. At first I thought it was web-responsive but turns out not (I tested with database pre-web-responsive to make sure database wasn't borked). Anyway, I ended up getting it to work with having only kanban field being display_name. Very strange, but I spent too much time on this and didn't want to throw anymore at this.

Since I don't have any specific need, and in the interest of other areas where help may be required, I'm closing this. Thank you for all your help @pedrobaeza in moving this forward.
